src/main/java/com/example/demo/service/impl/BookServiceImpl.java

```diff
@@ -24,13 +25,29 @@ public BookDto save(CreateBookRequestDto bookDto) {
     @Override
     public List<BookDto> findAll() {
         return bookRepository.findAll().stream()
-                .map(bookMapper::toDto)
-                .toList();
+            .map(bookMapper::toDto)
+            .toList();
     }
 
     @Override
     public BookDto getById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
-                () -> new EntityNotFoundException("Can not get book by id:" + id)));
+            () -> new EntityNotFoundException("Can not get book by id:" + id)));
+    }
+
+    @Override
+    public BookDto update(CreateBookRequestDto createBookRequestDto, Long id) {
+        if (getById(id) != null) {
```

Comment:
we'll get error when entity to update is not found
```suggestion
        if (getById(id) == null) {
```